ignite/contrib/handlers/trains_logger.py

```diff
@@ -659,32 +661,84 @@ def _setup_check_trains(self, logger, output_uri):
         if output_uri:
             self._task.output_uri = output_uri
 
+    class _CallbacksContext:
```

Comment:
@jkhenning as said previously (https://github.com/pytorch/ignite/pull/1135#issuecomment-645045035), we need to cover `_CallbacksContext.pre_callback` and `_CallbacksContext.post_callback` with basic unit tests to ensure the correct `model_info` as output.

Reply:
@vfdev-5 Since the unit tests don't initialize a Trains Task, we can't really get a real callback from Trains and so can't test the `model_info` type returned by Trains. The instance returned to Trains is never instantiated in the Ignite code and we always return the same instance we received from Trains, so I don't see what can be tested there. Also, since that type is a Trains internal type, I'm not sure I understand what can be tested in the context of an Ignite unit test...
Be that as it may, I might be missing something :) - can you please explain what you meant?

Reply:
I see what you mean, however, what can be tested is (maybe a bit artificial) that 
`model_info.upload_filename` and `model_info.local_model_id` are correct as well as the content of `_slots`when returned from `pre_callback`. For example, first time model is saved, there is a `model_info` passed to `pre_callback` and we can check the fields of output `model_info`.
And something similar about other fields of `model_info` from `post_callback`.
What do you think ?

Reply:
I see your point, however the thing is we currently don't have a way to get Trains to call these callbacks in a unit-test scenario, so we'll have to create a unit test that will:
* Mimic what `TrainsSaver.__call__()` usually does, i.e. create a `_CallbacksContext` instance
* Create a `model_info`-like object with the required attributes
* Pass it to `pre_callback()`
* Pass it to `post_callback()`

It's doable, but I have a feeling all we'll be testing is that the unit-test code is tailored exactly to the callbacks code to make sure they don't fail 😢 

However, I'll be happy to go with whatever you decide 😄

Reply:
👍 

> It's doable, but I have a feeling all we'll be testing is that the unit-test code is tailored exactly to the callbacks code to make sure they don't fail 

Yes, it will be rather redundant test to ensure that returned structures are good, but it is necessary especially taking into account this PR #1127 where user would like to change filename pattern.